nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-file-authorizer/src/main/java/org/apache/nifi/authorization/FileUserGroupProvider.java

```diff
@@ -455,36 +466,92 @@ public void setNiFiProperties(NiFiProperties properties) {
+    }
+
+    @Override
+    public synchronized void forciblyInheritFingerprint(final String fingerprint) throws AuthorizationAccessException {
+        if (fingerprint == null || fingerprint.trim().isEmpty()) {
+            logger.info("Inheriting Empty Users & Groups. Will backup existing Uesrs & Groups first.");
+            backupUsersAndGroups();
+            purgeUsersAndGroups();
+
+            return;
+        }
+
+        final UsersAndGroups usersAndGroups = parseUsersAndGroups(fingerprint);
+
+        if (isInheritable(usersAndGroups)) {
+            logger.debug("Inheriting cluster's Users & Groups");
+            inherit(usersAndGroups);
+        } else {
+            logger.info("Cannot directly inherit proposed Users & Groups so will backup existing Users & Groups and then replace with proposed configuration");
+            backupUsersAndGroups();
+            purgeUsersAndGroups();
+            addUsersAndGroups(usersAndGroups);
+        }
+    }
+
+    public void backupUsersAndGroups() throws AuthorizationAccessException {
+        final UserGroupHolder holder = userGroupHolder.get();
+        final Tenants tenants = holder.getTenants();
+
+        final DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss");
+        final String timestamp = dateFormat.format(new Date());
+        final File destinationFile = new File(tenantsFile.getParentFile(), tenantsFile.getName() + "." + timestamp);
+        logger.info("Writing backup of Users & Groups to {}", destinationFile.getAbsolutePath());
+
+        try {
+            saveTenants(tenants, destinationFile);
+        } catch (final JAXBException jaxb) {
+            throw new AuthorizationAccessException("Could not backup existing Users and Groups so will not inherit new Users and Groups", jaxb);
+        }
+    }
+
+    public synchronized void purgeUsersAndGroups() {
+        final UserGroupHolder holder = userGroupHolder.get();
+        final Tenants tenants = holder.getTenants();
+
+        tenants.setGroups(new Groups());
+        tenants.setUsers(new Users());
+
+        saveAndRefreshHolder(tenants);
     }
 
     @Override
     public void checkInheritability(String proposedFingerprint) throws AuthorizationAccessException {
+        final UsersAndGroups proposedUsersAndGroups;
         try {
             // ensure we understand the proposed fingerprint
-            parseUsersAndGroups(proposedFingerprint);
+            proposedUsersAndGroups = parseUsersAndGroups(proposedFingerprint);
         } catch (final AuthorizationAccessException e) {
             throw new UninheritableAuthorizationsException("Unable to parse the proposed fingerprint: " + e);
         }
 
-        final UserGroupHolder usersAndGroups = userGroupHolder.get();
-
         // ensure we are in a proper state to inherit the fingerprint
-        if (!usersAndGroups.getAllUsers().isEmpty() || !usersAndGroups.getAllGroups().isEmpty()) {
+        if (!isInheritable(proposedUsersAndGroups)) {
```

Comment:
1) `proposedUsersAndGroups` are not used in `isInheritable`.
2) Does it make sense to still have this check if we support force inheriting? Failing on parse makes sense but not sure about this one.